docs/todo-due-dates.md

```diff
@@ -0,0 +1,178 @@
+
+## Workflows
+
+### Normal todo workflow
+
+- No due date is expected when adding todo items with no threshold provided in the `package.json` or the CLI arguments `daysToError` or `daysToWarn`.
+
+### Todos that are past-due become errors
+
+Example use case: 20 days for todos to be considered errors since its creation date.
+
+**Via config:**
+
+In `package.json`:
+
+ ```json
+{
+  "lintTodo": {
+    "daysToError": 20,
+  }
+}
+```
+
+**Via `eslint` CLI:**
+
+```
+# to configure todos to transition to errors 20 days after creation
+UPDATE_TODO=1 DAYS_TO_ERROR_TODO=20 yarn eslint . --format eslint-formatter-todo
+```
+
+**Via `ember-template-lint` CLI:**
+
+```
+# to configure todos to transition to errors 20 days after creation
+yarn ember-template-lint . --update-todo --days-to-error-todo 20
+```
+
+### Todos that are past-due become warnings, then errors
+
+Use case: give 20 days for todos to be considered `warning`s, then 5 additional days until they are considered `error`s.
+
+**Via config:**
+
+```json
+{
+  "lint-todo": {
+    "daysToWarn": 20,
+    "daysToError": 25
+  }
+}
+```
+
+**Via `eslint` CLI:**
+
+```
+# to configure todos to transition to warnings 20 days after creation and to errors 25 days after creation
+UPDATE_TODO=1 DAYS_TO_WARN_TODO=20 DAYS_TO_ERROR_TODO=25 yarn eslint . --format eslint-formatter-todo
+```
+
+**Via `ember-template-lint` CLI:**
+
+```
+# to configure todos to transition to warnings 20 days after creation and to errors 25 days after creation
+yarn ember-template-lint . --update-todo --days-to-warn-todo 20 --days-to-error-todo 25
+```
+
+### Invalid option: only `daysToWarn` is present
+
+If `daysToWarn` is passed without `daysToError` an error must be produced.
+
+**Via `eslint` CLI:**
+
+```
+$ UPDATE_TODO=1 DAYS_TO_WARN_TODO=20 yarn eslint . --format eslint-formatter-todo
+
+Error: `DAYS_TO_WARN_TODO` must be used with `DAYS_TO_ERROR_TODO`.
+```
+
+**Via `ember-template-lint` CLI:**
+
+```
+$ yarn ember-template-lint . --update-todo --days-to-warn-todo 20
+
+Error: `--days-to-warn-todo` must be used with `--days-to-error-todo`.
+```
+
+#### Invalid option: `daysToWarn` is greater or equal than `daysToError`
+
+If `daysToWarn` is greater or equal than `daysToError` an error must be produced:
+
+**Via `eslint` CLI:**
+
+```
+$ UPDATE_TODO=1 DAYS_TO_WARN_TODO=20 DAYS_TO_ERROR_TODO=5 yarn eslint . --format eslint-formatter-todo
+
+Error: `DAYS_TO_WARN_TODO` value must be less than `DAYS_TO_ERROR_TODO`.
+```
+
+**Via `ember-template-lint` CLI:**
+
+```
+$ yarn ember-template-lint . --update-todo --days-to-warn-todo 20 --days-to-error-todo=5
+
+Error: `--days-to-warn-todo` value must be less than `--days-to-error-todo`.
+```
+
+
+## Proposed changes to the schema to include due date
+
+```ts
+interface TodoData {
+  ...
+  createdDate: Date;
+  daysToWarn?: int;
+  daysToError?: int;
```

Comment:
I feel like these should be actual dates, not offsets. Then this could become:

```suggestion
  warnDate?: Date;
  errorDate?: Date;
```

Reply:
We initially wanted to store the offset so we wouldn't be serializing hard dates in the todos. That said, this is simply pre-calculating the date using the provided start date and offset, so I think this is fine. This also has the added benefit of being more readable.